framework/src/main/java/org/checkerframework/framework/type/TypeFromMemberVisitor.java

```diff
@@ -95,7 +95,9 @@ public AnnotatedTypeMirror visitVariable(VariableTree variableTree, AnnotatedTyp
                     result.replaceAnnotation(anno);
                 } else {
                     // Type annotations apply to the innermost type.
-                    innerType.replaceAnnotation(anno);
+                    // We need to use addAnnotation instead of replaceAnnotation, otherwise testcase
```

Comment:
Instead of adding this comment to all three of these replace -> add changes, add a comment to the class. Don't focus on the test failure, but the reason why this makes sense, similar to what it now says in the PR description.